File: Safari-609.1.11.1/LayoutTests/http/tests/security/resources/credentials-in-referer-frame.php
<?php
 if (!isset($_SERVER['PHP_AUTH_USER'])) {
   header('WWW-Authenticate: Basic realm="WebKit test - credentials-in-referer"');
   header('HTTP/1.0 401 Unauthorized');
   echo 'Authentication canceled';
   exit;
 }
?>
<script>
if (window.testRunner) {
    testRunner.waitUntilDone();
    testRunner.dumpAsText();
}

function log(message)
{
    parent.document.getElementById("log").innerHTML += message + "<br>";
}

window.onload = function() {
    var xhr = new XMLHttpRequest;
    xhr.open("GET", "credentials-in-referer.php", false);
    xhr.send(null);
    log("Sync XHR: " + (xhr.responseText.match(/FAIL/) ? "FAIL" : "PASS"));

    xhr.open("GET", "credentials-in-referer.php", true);
    xhr.send(null);
    xhr.onload = onXHRLoad;
}

function onXHRLoad(evt)
{
    log("Async XHR: " + (evt.target.responseText.match(/FAIL/) ? "FAIL" : "PASS"));
    log("DONE");
    if (window.testRunner)
        testRunner.notifyDone();
}
</script>
<script src="credentials-in-referer.php"></script>

File: LayoutTests/http/tests/security/resources/credentials-in-referer-frame.py
#!/usr/bin/env python3

import base64
import os
import sys

username = base64.b64decode(os.environ.get('HTTP_AUTHORIZATION', ' Og==').split(' ')[1]).decode().split(':')[0]

sys.stdout.write('Content-Type: text/html\r\n')

if not username:
    sys.stdout.write(
        'WWW-Authenticate: Basic realm="WebKit test - credentials-in-referer"\r\n'
        'status: 401\r\n\r\n'
        'Authentication canceled'
    )
    sys.exit(0)

sys.stdout.write(
    '\r\n<script>\n'
    'if (window.testRunner) {\n'
    '    testRunner.waitUntilDone();\n'
    '    testRunner.dumpAsText();\n'
    '}\n'
    '\n'
    'function log(message)\n'
    '{\n'
    '    parent.document.getElementById("log").innerHTML += message + "<br>";\n'
    '}\n'
    '\n'
    'window.onload = function() {\n'
    '    var xhr = new XMLHttpRequest;\n'
    '    xhr.open("GET", "credentials-in-referer.py", false);\n'
    '    xhr.send(null);\n'
    '    log("Sync XHR: " + (xhr.responseText.match(/FAIL/) ? "FAIL" : "PASS"));\n'
    '\n'
    '    xhr.open("GET", "credentials-in-referer.py", true);\n'
    '    xhr.send(null);\n'
    '    xhr.onload = onXHRLoad;\n'
    '}\n'
    '\n'
    'function onXHRLoad(evt)\n'
    '{\n'
    '    log("Async XHR: " + (evt.target.responseText.match(/FAIL/) ? "FAIL" : "PASS"));\n'
    '    log("DONE");\n'
    '    if (window.testRunner)\n'
    '        testRunner.notifyDone();\n'
    '}\n'
    '</script>\n'
    '<script src="credentials-in-referer.py"></script>\n'
)